Flags locales

wyattjoh opened the pull request:
## What does this PR do?

- Flags now are all required to pre-define their types in the graph, or at the very least provide their translations in locales.

Ref: https://www.pivotaltracker.com/story/show/151179316

## How do I test this PR?

1. Create some flags
2. Check out the translated responses in the admin

wyattjoh committed:
Initial work for locking flag type and providing translations

wyattjoh committed:
cleaned up graph mutators

wyattjoh committed:
changed custom error to standardized error

wyattjoh committed:
Merge branch 'master' into flags-locales

wyattjoh committed:
updated translations

wyattjoh committed:
added toxic comment translations

kgardnr committed:
Merge branch 'master' into flags-locales

alinhle commented on graph/typeDefs.graphql:
```diff
@@ -948,7 +972,7 @@ input CreateFlagInput {
 
   # The reason for flagging the item.
-  reason: String!
+  reason: FLAG_REASON
```
Required?

wyattjoh replied:
Nope!

alinhle commented:
We also need to adapt toxic comment for its TOXIC_COMMENT flag.

alinhle reviewed (changes requested):
![image](https://user-images.githubusercontent.com/14221600/31288532-992b3afc-aac5-11e7-924b-1ff56589c719.png)
The graphql refuses to load any system flags whose reasons are not defined in the `FLAG_REASON` type, e.g. trust, suspect and toxic comment flags

wyattjoh committed:
Merge branch 'master' into flags-locales

wyattjoh committed:
Graph fixes and migrations

- Added migrations for mapping old keys
- Added migration for fixing old dont agree flags
- Removed enum requirement for output

kgardnr committed:
Merge branch 'master' into flags-locales

wyattjoh committed:
small naming change

wyattjoh committed:
Merge branch 'master' into flags-locales

wyattjoh committed:
Merge branch 'master' into flags-locales

kgardnr committed:
Merge branch 'master' into flags-locales

kgardnr committed:
Merge branch 'master' into flags-locales

kgardnr committed:
Merge branch 'master' into flags-locales

wyattjoh committed:
Merge branch 'master' into flags-locales

alinhle committed:
Merge branch 'master' into flags-locales

alinhle committed:
Merge branch 'master' into flags-locales

alinhle committed:
Restore suspect label

wyattjoh committed:
Merge branch 'master' into flags-locales

kgardnr committed:
Merge branch 'master' into flags-locales

wyattjoh committed:
Merge branch 'master' into flags-locales

wyattjoh committed:
Merge branch 'master' into flags-locales

wyattjoh commented:
Future PR will merge this branch in.
